protobuf/lib/src/protobuf/proto3_json.dart

```diff
@@ -89,11 +100,32 @@ Object? _writeToProto3Json(_FieldSet fs, TypeRegistry typeRegistry) {
   var result = <String, dynamic>{};
   for (var fieldInfo in fs._infosSortedByTag) {
     var value = fs._values[fieldInfo.index!];
-    if (value == null || (value is List && value.isEmpty)) {
+    var overrideForEmitsDefaults = false;
+    dynamic overrideForEmitsDefaultsValue;
+    if (context.emitDefaults) {
+      if (fieldInfo.isRepeated && isNullOrEmptyList(value)) {
+        overrideForEmitsDefaults = true;
+        overrideForEmitsDefaultsValue = [];
+      } else if (fieldInfo.isMapField && isNullOrEmptyMap(value)) {
+        overrideForEmitsDefaults = true;
+        overrideForEmitsDefaultsValue = {};
+      } else if (_isBytes(fieldInfo.type) && isNullOrEmptyList(value)) {
```

Comment:
All those `isNullOrEmptyList` etc. functions are actually testing for default values, so I think a better naming would be `isDefaultListField`, `isDefaultMapField` or something along those lines. Does this make sense?

Not a problem in this PR, but it's a bit unfortunate that we represent defaults of a field with multiple values, e.g. `null` or an empty list for repeated fields.

Reply:
Good idea.  This has been fixed.  Thanks.